src/cryptography/hazmat/primitives/kdf/scrypt.py

```diff
@@ -56,9 +56,20 @@ def derive(self, key_material):
 
         if not isinstance(key_material, bytes):
             raise TypeError("key_material must be bytes.")
-        return self._backend.derive_scrypt(
-            key_material, self._salt, self._length, self._n, self._r, self._p
-        )
+
+        try:
+            return self._backend.derive_scrypt(
+                key_material, self._salt, self._length,
+                self._n, self._r, self._p
+            )
+        except InternalError as e:
```

Comment:
This is being handled in the wrong place, we should handle this inside `derive_scrypt` instead of raising an `InternalError.`

Reply:
Ah, I see. Thanks!